proto/structured.proto

```diff
@@ -104,3 +105,234 @@ message CreateTableResponse {
+  // aborted as conflicting transactions or non-transactional commands
+  // with UserPriority=1. This value is ignored if Txn is
+  // specified. If neither this value nor Txn is specified, the value
+  // defaults to 1.
+  optional int32 user_priority = 6 [default = 1];
+  // Txn is set non-nil if a transaction is underway. To start a txn,
+  // the first request should set this field to non-nil with name and
+  // isolation level set as desired. The response will contain the
+  // fully-initialized transaction with txn ID, priority, initial
+  // timestamp, and maximum timestamp. The transaction returned in a
+  // response is feed to the next request.
+  optional Transaction txn = 7;
+  // ReadConsistency specifies the consistency for read
+  // operations. The default is CONSISTENT. This value is ignored for
+  // write operations.
+  optional ReadConsistencyType read_consistency = 8 [(gogoproto.nullable) = false];
+}
+
+// TableResponseHeader is returned with every structured API response.
+message TableResponseHeader {
+  // Error is non-nil if an error occurred.
+  optional Error error = 1;
+  // Timestamp specifies time at which read or write actually was
+  // performed. In the case of both reads and writes, if the timestamp
+  // supplied to the request was 0, the wall time of the node
+  // servicing the request will be set here. Additionally, in the case
+  // of writes, this value may be increased from the timestamp passed
+  // with the RequestHeader if the key being written was either read
+  // or written more recently.
+  //
+  // TODO(vivek): Is this only relevant in the non-transactional case?
+  optional Timestamp timestamp = 2 [(gogoproto.nullable) = false];
+  // Transaction is non-nil if the request specified a non-nil
+  // transaction. The transaction timestamp and/or priority may have
+  // been updated, depending on the outcome of the request.
+  optional Transaction txn = 3;
+}
+
+// A ColumnCollection represents a collection of columns as name:value pairs.
+message ColumnCollection {
+  message ColumnNameValue {
+    // Column name.
+    optional string name = 1;
+    optional MVCCValue value = 2 [(gogoproto.nullable) = false];
+  }
+  repeated ColumnNameValue columns = 1;
+}
+
+// A ValueCollection is a collection of MVCC values.
+message ValueCollection {
+  repeated MVCCValue values = 1;
+}
+
+// A GetTableRowRequest is used to read a table row.
+message GetTableRowRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ColumnCollection key = 2 [(gogoproto.nullable) = false];
+  // Get specified columns from the row.
+  repeated string columns = 3;
+}
+
+// A GetTableRowResponse returns a row.
+// If the key doesn't exist, it returns an empty row.
+message GetTableRowResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // A deleted column has deleted set.
```

Comment:
I see now why you're using `MVCCValue`, but really you want to be using `Value` and the existing semantics that a `nil` `Bytes` and `Integer` field indicates the value does not exist.